App/SB/views/ThreadsDetail/index.js

```diff
@@ -174,29 +206,51 @@ const mapStateToProps = (state, ownProps) => {
       ? 'Any new photos you take will be added to your Textile wallet.'
       : 'Share your first photo to the ' + threadName + ' thread.')
 
+
+  // A little bit of feedback for the user to show that an image is
+  // processing... fills the gap before it shows up in the thread
+  const pendingShares = state.cameraRoll.pendingShares[threadId] || []
+  let progress = 0.0
+  if (pendingShares.length > 0) {
+    const firstShare = pendingShares[0]
+    if (firstShare.caption) {
+      progress = 0.3
+      if (firstShare.addResult) {
+        progress = 0.6
+        if (state.uploadingImages.images[firstShare.addResult.id]) {
+          progress = 0.8
+        }
+      }
+    }
+  }
```

Comment:
Cool. The first two steps are almost instantaneous, right? I wonder if just displaying the actual image upload progress would be more meaningful (kinda like our pie views before).

Reply:
this is pre-upload... the image even shows in the thread before remote pin, so no concrete requirement to show pie. so if you are looking at the ui there is 1 second where nothing appears, it feels a lot better to see progress, otherwise it just feels a bit laggy for no stated reason. give it a try

Reply:
nice, makes sense.